Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -797,6 +825,61 @@ AXTextMarker AXTextMarker::findMarker(AXDirection direction, AXTextUnit textUnit
         }
         return linePosition;
     }
+
+    if (textUnit == AXTextUnit::Word) {
+        unsigned offset = this->offset();
+        AXTextMarker resultMarker = *this;
+
+        String flattenedRuns = currentRuns->substring(0, currentRuns->totalLength());
+
+        // objectBorder maintains the position in flattenedRuns between the current object's text and the previously scanned object(s)
+        int objectBorder = direction == AXDirection::Next ? 0 : flattenedRuns.length();
+        while (currentObject) {
+            int start, end;
+            findWordBoundary(flattenedRuns, offset, &start, &end);
+            if (direction == AXDirection::Previous) {
+                // allow previous word to start at text marker if we are in the middle of the word
+                // check that end is within the word using the wordLength
+                int previousWordStart = findNextWordFromIndex(flattenedRuns, offset, false);
+                int end;
```

Comment:
Do we need to re-declare `int end` here, or can we just rely on its declaration above?

Reply:
We might be able to rely on the above, let me see what happens if I remove it

Reply:
Yep, it's totally fine to remove this